src/dmd/builtin.d

```diff
@@ -259,45 +253,27 @@ extern (C++) Expression eval_bsr(Loc loc, FuncDeclaration fd, Expressions* argum
     uinteger_t n = arg0.toInteger();
     if (n == 0)
         error(loc, "`bsr(0)` is undefined");
-    int k = 0;
-    while (n >>= 1)
-    {
-        ++k;
-    }
-    return new IntegerExp(loc, k, Type.tint32);
+    return new IntegerExp(loc, core.bitop.bsr(n), Type.tint32);
 }
 
 extern (C++) Expression eval_bswap(Loc loc, FuncDeclaration fd, Expressions* arguments)
 {
     Expression arg0 = (*arguments)[0];
     assert(arg0.op == TOK.int64);
     uinteger_t n = arg0.toInteger();
-    enum BYTEMASK = 0x00FF00FF00FF00FFL;
-    enum SHORTMASK = 0x0000FFFF0000FFFFL;
-    enum INTMASK = 0x0000FFFF0000FFFFL;
-    // swap adjacent ubytes
-    n = ((n >> 8) & BYTEMASK) | ((n & BYTEMASK) << 8);
-    // swap adjacent ushorts
-    n = ((n >> 16) & SHORTMASK) | ((n & SHORTMASK) << 16);
     TY ty = arg0.type.toBasetype().ty;
-    // If 64 bits, we need to swap high and low uints
     if (ty == Tint64 || ty == Tuns64)
-        n = ((n >> 32) & INTMASK) | ((n & INTMASK) << 32);
-    return new IntegerExp(loc, n, arg0.type);
+        return IntegerExp(loc, core.bitop.bswap(cast(ulong) n), arg0.type);
+    else
+        return IntegerExp(loc, core.bitop.bswap(cast(uint) n), arg0.type);
 }
 
 extern (C++) Expression eval_popcnt(Loc loc, FuncDeclaration fd, Expressions* arguments)
 {
     Expression arg0 = (*arguments)[0];
     assert(arg0.op == TOK.int64);
     uinteger_t n = arg0.toInteger();
-    int cnt = 0;
-    while (n)
-    {
-        cnt += (n & 1);
-        n >>= 1;
-    }
-    return new IntegerExp(loc, cnt, arg0.type);
```

Comment:
This type for the result of `eval_popcnt` -- `arg0.type` -- was incorrect.  The result of `popcnt` is always `int`. See:

https://github.com/dlang/druntime/blob/db2910ee7425f91dc6fe38f149f4b8eaa3b88f7f/src/core/bitop.d#L561
https://github.com/dlang/druntime/blob/db2910ee7425f91dc6fe38f149f4b8eaa3b88f7f/src/core/bitop.d#L593